### ISSUE TITLE
Arguments are not validated with Generic Extension Methods

### ISSUE THREAD

**Issue by iamkrillin** [dotnet/roslyn#15837]
**Version Used**: 1.3.1.60616 (vs 2015 update 3)

**Steps to Reproduce**:

1.  Create an extension method with a generic argument
2.  Call extension method without type specified

**Expected Behavior**:
Compiler Error and error highlight in the IDE

**Actual Behavior**:
CS7038 Failed to emit module, no error highlighting

The following code demonstrates the problem:

```
namespace ConsoleApplication3
{
    class Program
    {
        static void Main(string[] args)
        {
            int id = 1.GetItem<int>();
            object item = 1.GetItem<>();
        }
    }

    public static class Extensions
    {
        public static T GetItem<T>(this object item)
        {
            return default(T);
        }
    }
}
```

**Comment by gafter** [comment]
I think this may have been fixed recently.

**Comment by jcouv** [comment]
I tried it just now ([TryRsolyn](http://tryroslyn.azurewebsites.net/#b:master/HYQwtgpgzgDiDGEAEBhA9sKaA2ECCMM2AlvCAC7EYDMAUAN61LNLzYhRRIAKATmgHNe4Ji0YsJSKOQqkkANzTEAJkgCyIYsAAU03loEBtALpIQvAVACUoyUnF3JW8khVIAvEgCMAOgDiEOQAkuQQYAA8zgB82lYA3LaOzGgARgBWEPAuxKFgHt7+gSFh4THxiRIAvonViTAArikk8FIylC1sHFwAogAeoZhUmIkOdg1NctKyLQAqSAHBueEzMeQAFsRcqRlZrrk2SaNJSLyB9bzASMoQAGYg9djk2jPlSdVVtJVAA===)) and got a proper diagnostic:

`(7,26,7,37): error CS0305: Using the generic method group 'GetItem' requires 1 type arguments`

This was most likely fixed by https://github.com/dotnet/roslyn/pull/15966

@OmarTawfik can you confirm this seems the same issue you fixed and close?

**Comment by OmarTawfik** [comment]
Yes. Fixed in #15966, and a test for that case is added in https://github.com/dotnet/roslyn/blob/49415ad4794c95265007a80a5fd9e1f519509dd9/src/Compilers/CSharp/Test/Semantic/Semantics/BindingTests.cs#L3430
